app/src/main/java/com/hanseltritama/connectthree/MainActivity.java

```diff
@@ -1,9 +1,123 @@
 package com.hanseltritama.connectthree;
 
+import android.content.DialogInterface;
+import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageButton;
+import android.widget.Toast;
 
 public class MainActivity extends AppCompatActivity {
+    private int turn = 0;
+    private int player = 1;
+    private int[][] board = new int[3][3];
+
+    public void putRedToken(ImageButton imageButton) {
+        imageButton.setImageDrawable(getResources().getDrawable(R.drawable.red));
+        imageButton.setAlpha(1f);
+    }
+
+    public void putYellowToken(ImageButton imageButton) {
+        imageButton.setImageDrawable(getResources().getDrawable(R.drawable.yellow));
+        imageButton.setAlpha(1f);
+    }
+
+    public int getXCoord(int imgBtnId) {
+        switch (imgBtnId) {
+            case 0:
+            case 1:
+            case 2: return 0;
+            case 3:
+            case 4:
+            case 5: return 1;
+            case 6:
+            case 7:
+            case 8: return 2;
+            default: return 0;
+        }
+    }
+
+    public int getYCoord(int imgBtnId) {
+        switch (imgBtnId) {
+            case 0:
+            case 3:
+            case 6: return 0;
+            case 1:
+            case 4:
+            case 7: return 1;
+            case 2:
+            case 5:
+            case 8: return 2;
+            default: return 0;
+        }
+    }
+
+    public boolean checkCol(int[][] board, int x_coord) {
+        return (board[0][x_coord] == board[1][x_coord]) && (board[1][x_coord] == board[2][x_coord]);
+    }
+
+    public boolean checkRow(int[][] board, int y_coord) {
+        return (board[y_coord][0] == board[y_coord][1]) && (board[y_coord][1] == board[y_coord][2]);
+    }
+
+    public boolean checkDiagonalLeft(int[][] board, int x_coord, int y_coord) {
+        return (x_coord == y_coord) && ((board[0][0] == board[1][1]) && (board[1][1] == board[2][2]));
```

Comment:
not sure if the diagonal logic correct....what if x_coordinate is 2 and y_coordinate is 0?

Reply:
I put `checkDiagonalRight` function as well to address that case